src/compiler/diagnosticMessages.json

```diff
@@ -1974,7 +1974,7 @@
         "category": "Error",
         "code": 6053
     },
-    "File '{0}' must have extension '.ts' or '.d.ts'.": {
+    "File '{0}' has unsupported extension. The only supported extensions are {1}.": {

```

Comment:
I would rather switch back to the old message then, at least it had the "or".. :D